### DISCUSSION TITLE
Hung on "configuration complete"

### DISCUSSION THREAD

**Discussion by Tegguy** [tagyoureit/nodejs-poolController#705] [App and O/S Config Questions]
I'm working to try to setup NJSPC on a raspberry pi 3 and I need some help. I got everything done and went to run the NPM Start command on the controller but my command prompt is stuck a "configuration complete" and doesn't return to the line to allow me to enter another command. 

Any suggestions?

![image](https://user-images.githubusercontent.com/83373401/229134153-d6c01fcb-5a54-4ece-b668-d9442bb0481c.png)

**Comment by rstrouse** [comment]
You cannot enter commands in the nodejs console.  What are you trying to do?

**Reply by Tegguy** [reply]
I ran the NPM start command... I was looking to change directory and start the dashpanel however, I never got the line to allow me to enter the change directory command. I wasn't sure if this was normal behavior or not. Following the guide on troublefreepool his went from "configuration complete" to returning to the line to enter commands (sorry not sure what it's called)

Sorry let me clarify I was telneted into my raspberry pi's command line on the screenshot above I provided

**Reply by tagyoureit** [reply]
[PM2](https://github.com/tagyoureit/nodejs-poolController/wiki/Automatically-start-at-boot---PM2-&-Systemd) is your best option for running these in the background.  They are designed to stay running so will never return to a prompt.

Alternatively, for testing, you can start a 2nd telnet session.  However, once the connection is severed the telnet session will terminate and kill any running apps.

**Reply by dcorrea78** [reply]
I just installed this last night and noticed the same behavior but it is normal. I'm not sure how old the screenshots are in the TFP guide, but when you run the npm start command, you are starting the application. You will not get back to the command prompt unless the app is closed or terminated.

Basically when following the guide, you need to open two different CLI windows. One to start and run the poolcontroller and another one to run the dashboard, at least until you setup the auto start scripts which is where I'm currently struggling.

**Reply by Tegguy** [reply]
> I just installed this last night and noticed the same behavior but it is normal. I'm not sure how old the screenshots are in the TFP guide, but when you run the npm start command, you are starting the application. You will not get back to the command prompt unless the app is closed or terminated.
> 
> Basically when following the guide, you need to open two different CLI windows. One to start and run the poolcontroller and another one to run the dashboard, at least until you setup the auto start scripts which is where I'm currently struggling.

Thanks I figured the same thing out. I was able to get the auto start to work. What are you struggling with?

**Comment by dcorrea78** [comment]
Nothing now. I was struggling to get pm2 to properly autostart, but I updated the pm2 package this morning and got it working.
